src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractSyncNamespaceService.MoveFileCodeAction.cs

```diff
@@ -0,0 +1,187 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CodeRefactorings.SyncNamespace
+{
+    internal abstract partial class AbstractSyncNamespaceService<TNamespaceDeclarationSyntax, TCompilationUnitSyntax>
+        where TNamespaceDeclarationSyntax : SyntaxNode
+        where TCompilationUnitSyntax : SyntaxNode 
+    {
+        internal sealed class MoveFileCodeAction : CodeAction
+        {
+            private readonly State _state;
+            private readonly ImmutableArray<string> _newfolders;
+
+            public override string Title
+                => _newfolders.Length > 0 
+                ? string.Format(FeaturesResources.Move_file_to_0_folder, string.Join(PathUtilities.DirectorySeparatorStr, _newfolders)) 
+                : FeaturesResources.Move_file_to_project_root_folder;
+
+            public MoveFileCodeAction(State state, ImmutableArray<string> newFolders)
+            {
+                _state = state;
+                _newfolders = newFolders;
+            }
+
+            internal override bool PerformFinalApplicabilityCheck => true;
+
+            internal override bool IsApplicable(Workspace workspace)
+            {
+                // Due to some existing issue, move file action is not available for CPS projects.
+                return workspace.CanRenameFilesDuringCodeActions(workspace.CurrentSolution.GetDocument(_state.OriginalDocumentId).Project);
+            }
+
+            protected override Task<IEnumerable<CodeActionOperation>> ComputeOperationsAsync(CancellationToken cancellationToken)
+                => MoveFileToMatchNamespaceAsync(cancellationToken);
```

Comment:
is there a reason this just calls to another method and does nothing else?  why have both methods?